R/zchunk_L251.en_ssp_nonco2.R

```diff
@@ -35,106 +40,214 @@ module_emissions_L251.en_ssp_nonco2_DISABLED <- function(command, ...) {
+          tidyr::nesting(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology, agg_sector),
+          fill = list(value = NA)) %>%
+      mutate(year = as.integer(year))  %>%
+      group_by(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology, agg_sector) %>%
+      mutate(value = approx_fun(year, value))  %>%
+      # Now add regional information.
+      left_join_error_no_match(A_regions, by = c("GCAM_region_ID"))  %>%
+      # Emission coefficients values are too long, so we round to the 10th decimal point.
+      mutate(emiss.coeff = round(value, 10)) %>%
+      ungroup() %>%
+      # Discard columns that are not needed.
+      select(-MAC_region, -bio_N2O_coef, -SO2_name, -GAINS_region, -GCAM_region_ID, -agg_sector, -value) ->
+    L251.ssp2_ef
+
+    L161.SSP34_EF %>%
+      # Input data only contains 3 future years.  Need to include years between 2010-2100 in 5 year segments,
+      # and interpolate between each time segment.
+      tidyr::complete(year = SSP_FUTURE_YEARS,
+          tidyr::nesting(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology, agg_sector),
+          fill = list(value = NA)) %>%
+      mutate(year = as.integer(year))  %>%
+      group_by(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology, agg_sector) %>%
+      mutate(value = approx_fun(year, value))  %>%
+      # Now add regional information.
+      left_join_error_no_match(A_regions, by = c("GCAM_region_ID"))  %>%
+      # Emission coefficients values are too long, so we round to the 10th decimal point.
+      mutate(emiss.coeff = round(value, 10)) %>%
+      ungroup() %>%
+      # Discard columns that are not needed.
+      select(-MAC_region, -bio_N2O_coef, -SO2_name, -GAINS_region, -GCAM_region_ID, -agg_sector, -value) ->
+    L251.ssp34_ef
+
+# Create a control table where year 2010 is the control period and renamed 1975, discarding all other years.
+     L251.ssp2_ef %>%
+      select(-emiss.coeff) %>%
+      subset(year == min(year)) %>%
+      mutate(year = 1975,
+             ctrl.name ="GDP_control") ->
+      L251.ctrl.delete
+
+# This section adds emissions controls for future years of vintaged technologies for SSP emission factors.
+     L251.ssp15_ef %>%
+       # Isolate the "electricity" supply sector.
+       subset(supplysector == "electricity") %>%
+       # Create 2 new columns for future emission factors.
+       # Future emission coefficient year is based on pre-existing "year" column.
+       # Future emission coefficient name is a descriptor, and is constant.
+       mutate(future.emiss.coeff.year = year,
+              future.emiss.coeff.name = "SSP_GAINS",
+              # Previous year column that only includes the electricity supply sector is now a constant.
+              year = 1975)->
+      L251.ssp15_ef_vin
+
+     L251.ssp2_ef %>%
+       # Isolate the "electricity" supply sector.
+       subset(supplysector == "electricity") %>%
```

Comment:
You should use `filter` for this operation and other similar operations throughout.